modules/social_features/social_user/social_user.install

```diff
@@ -206,3 +206,18 @@ function social_user_update_8810() {
     'view blocked user',
   ]);
 }
+
+/**
+ * Migrate navigation settings into the correct format.
+ */
+function social_user_update_8811() {
+  $config = \Drupal::configFactory()->getEditable('social_user.navigation.settings');
+  // If someone manually changed the value in the form then skip it.
+  if (!is_bool($config->get('display_my_groups_icon'))) {
+    $config->set('display_my_groups_icon', $config->get('display_my_groups_icon') === 1 ? TRUE : FALSE);
+  }
```

Comment:
I like the fact that we're being explicit but this may be the perfect place for type-casting :D 

```suggestion
  $config->set('display_my_groups_icon', (bool)$config->get('display_my_groups_icon'));
```

This even works for strings:
```php
echo '"1": '; var_dump((bool)"1");
echo '"0": '; var_dump((bool)"0");

echo '1: '; var_dump((bool)1);
echo '0: '; var_dump((bool)0);
```

outputs
```
"1": bool(true)
"0": bool(false)
1: bool(true)
0: bool(false)
```